media-libs/opencv/opencv343-3.4.3.recipe

```diff
@@ -0,0 +1,285 @@
+	lib:libopencv_objdetect$secondaryArchSuffix = $libVersionCompat
+	lib:libopencv_photo$secondaryArchSuffix = $libVersionCompat
+	lib:libopencv_shape$secondaryArchSuffix = $libVersionCompat
+	lib:libopencv_stitching$secondaryArchSuffix = $libVersionCompat
+	lib:libopencv_superres$secondaryArchSuffix = $libVersionCompat
+	lib:libopencv_video$secondaryArchSuffix = $libVersionCompat
+	lib:libopencv_videoio$secondaryArchSuffix = $libVersionCompat
+	lib:libopencv_videostab$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libavcodec$secondaryArchSuffix
+	lib:libavformat$secondaryArchSuffix
+	lib:libavresample$secondaryArchSuffix
+	lib:libavutil$secondaryArchSuffix
+	lib:libgdal$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libGLU$secondaryArchSuffix
+	lib:libgphoto2$secondaryArchSuffix
+	lib:libgstapp_1.0$secondaryArchSuffix
+	lib:libgstreamer_1.0$secondaryArchSuffix
+	lib:libgstvideo_1.0$secondaryArchSuffix
+	lib:libilmimf$secondaryArchSuffix #openexr
+	lib:libintl$secondaryArchSuffix
+	lib:libjasper$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:liblapack$secondaryArchSuffix
+	lib:libopenblas$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libQt5Concurrent$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Test$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libswscale$secondaryArchSuffix
+	lib:libtiff$secondaryArchSuffix
+	lib:libwebp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	opencv343${secondaryArchSuffix}_devel = $portVersionCompat
+	devel:libopencv_calib3d$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_core$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_dnn$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_features2d$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_flann$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_highgui$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_imgcodecs$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_imgproc$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_ml$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_objdetect$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_photo$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_shape$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_stitching$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_superres$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_video$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_videoio$secondaryArchSuffix = $libVersionCompat
+	devel:libopencv_videostab$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	opencv343$secondaryArchSuffix == $portVersion base
+	"
+CONFLICTS_devel="
+	opencv${secondaryArchSuffix}_devel
+	opencv342${secondaryArchSuffix}_devel
```

Comment:
@alaviss 
> I don't really know if mutual conflicts work, but if it does, then both packages just need to conflict opencv_devel.

I've just had a quick look at the conflicts in this recipe, and I can confirm that the `CONFLICTS` and `CONFLICTS_devel` are both correct, both in this recipe, `opencv343-3.4.3.recipe`, and in `opencv342-3.4.2.recipe`.
BTW, there are no conflcts in `opencv-3.4.1.recipe` and this is also 100% correct because it is enough to put the conflcts in the recipes with the most recent releases, `3.4.2` and `3.4.3`.
So my conclusion is that all conflcts are currently OK :)


Hmm, since we already have `3.4.1` and `3.4.2`, instead of adding `3.4.3` I wonder if it would not be easier to _bump_ `3.4.2` to `3.4.3`.
Well, I guess there are reasons for keeping `3.4.2`, so I'm OK with having 3 versions :)

Reply:
I don't wanted to bump every depending package... Lazyness, you know.

Reply:
All of our `opencv` version share the same `SONAME`, so you won't have to bump every package...